components/server/src/routes/report.py

```diff
@@ -40,7 +44,22 @@ def post_report_import(database: Database):
     """Import a preconfigured report into the database."""
     report = dict(bottle.request.json)
     report["delta"] = dict(uuids=[report["report_uuid"]])
-    result = import_json_report(database, report)
+
+    date_time = report_date_time()
+    data_model = latest_datamodel(database, date_time)
+
+    secret = database.secrets.find_one({"name": EXPORT_FIELDS_KEYS_NAME}, {"private_key": True, "_id": False})
+    private_key = secret["private_key"]
+
+    try:
+        decrypt_credentials(data_model, private_key, report)
+    except DecryptionError:
+        bottle.response.status = 400
+        bottle.response.content_type = "application/json"
+        return {"error": "Decryption of source credentials failed. Did you use the correct public key for encryption?"}
```

Comment:
I find the question "Did you use the correct public key for encryption?" a bit abstract. Suggestion: "Did you use the public key of this Quality-time instance to encrypt this report?"

Reply:
changed according to your suggestion.